### PR TITLE
chore(deps): Bump all dependencies (including kube and k8s-openapi)

### PR DESCRIPTION
**Pull request by sbernauer**
## Description

Sample operator PR: https://github.com/stackabletech/trino-operator/pull/458

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by Techassi** [comment]
I would argue that we change the title to "chore: Bump all dependencies".

File: Cargo.toml [hunk 1]
```diff
@@ -9,36 +9,36 @@ repository = "https://github.com/stackabletech/operator-rs"
...
 derivative = "2.2.0"
-tracing-opentelemetry = "0.19.0"
-opentelemetry = { version = "0.19.0", features = ["rt-tokio"] }
-opentelemetry-jaeger = { version = "0.18.0", features = ["rt-tokio"] }
+tracing-opentelemetry = "0.20.0"
+opentelemetry = { version = "0.20.0", features = ["rt-tokio"] }
+opentelemetry-jaeger = { version = "0.19.0", features = ["rt-tokio"] }
 stackable-operator-derive = { path = "stackable-operator-derive" }
-snafu = "0.7.4"
+snafu = "0.7.5"
```
Comment (Techassi) [inline Cargo.toml:37]: We currently depend on `thiserror` and `snafu`. I think we should get rid of `snafu` and use `thiserror` exclusively. In the PR https://github.com/stackabletech/stackable-cockpit/pull/55 I'm also replacing `snafu` with `thiserror` as it better integrates with other crates from the community.

Comment (sbernauer) [reply Cargo.toml:37]: No objections, I don't know them well enough. Is this crystal clear or should we discuss this in Arch meeting?

Comment (sbernauer) [reply Cargo.toml:37]: Have added it to the Arch meeting agenda. I would like to address this in a dedicated PR if that's ok for you

Comment (Techassi) [reply Cargo.toml:37]: > Is this crystal clear or should we discuss this in Arch meeting?

For me it is, but others might have some additional input on this...

> I would like to address this in a dedicated PR if that's ok for you

Yes, totally!

**Review by Techassi — APPROVED** [review]

**Pull request merged**